create_jenkins_job.py

```diff
@@ -121,7 +121,7 @@ def main(argv=None):
         'dont_notify_every_unstable_build': 'false',
         'build_timeout_mins': 0,
         'ubuntu_distro': 'jammy',
-        'el_release': '9',
+        'el_release': '9.2-pinned',
```

Comment:
I don't think it makes sense to use a custom value here instead of the true release we target. We didn't do this when we pinned a safe Jammy image and unless there's prior art or a strong motivation to deviate here I'd prefer not to.

Testing the upstream RHEL 9 build can be done by reverting this PR entirely and testing that branch.

Reply:
At the end of the day, I don't care too much.  I think it would be slightly nicer to be able to test whether RHEL-9/EPEL has rebuilt everything without a PR, but if we think that is too complicated I'm fine with just unconditionally pinning it.

Reply:
Left my thoughts on the comment below. If it's a blocker to merge it for you @nuclearsandwich I'm happy to change it to how it was before though I do see some value in adding an option here.

Reply:
I'm outnumbered and don't strongly object. So I'll not block on preference.